outputs.tf

```diff
@@ -1,31 +1,44 @@
 output "public_ip" {
   description = "Public IP of instance (or EIP )"
-  value       = "${coalesce(join("", aws_eip.default.*.public_ip), aws_instance.default.public_ip)}"
+  value       = "${join(",",coalescelist(concat(aws_eip.default.*.public_ip, list()),aws_instance.default.*.public_ip))}"
 }
 
 output "private_ip" {
   description = "Private IP of instance"
-  value       = "${join("", aws_instance.default.*.private_ip)}"
+  value       = "${join(",", aws_instance.default.*.private_ip)}"
 }
 
 output "private_dns" {
   description = "Private DNS of instance"
-  value       = "${join("", aws_instance.default.*.private_dns)}"
+  value       = "${join(",", aws_instance.default.*.private_dns)}"
 }
 
-output "public_dns" {
-  description = "Public DNS of instance (or DNS of EIP)"
-  value       = "${coalesce(join("", null_resource.eip.*.triggers.public_dns), aws_instance.default.public_dns)}"
-}
+# locals {
+#   eips_exist = "${signum(length(null_resource.eip.triggers)) == 1}"
+#   local_dns_output = "${local.eips_exist ? coalesce(concat(null_resource.eip.*.triggers.public_dns, list()), aws_instance.default.*.public_dns) : list() }"
+# }
+# output "public_dns" {
+#   description = "Public DNS of instance (or DNS of EIP)"
+#   value       = "${local.local_dns_output}"
+# }
 
 output "id" {
   description = "Disambiguated ID"
-  value       = "${join("", aws_instance.default.*.id)}"
+  value       = "${join(",", aws_instance.default.*.id)}"
 }
 
 output "ssh_key_pair" {
   description = "Name of used AWS SSH key"
-  value       = "${var.ssh_key_pair}"
+  value       = "${signum(length(var.ssh_key_pair)) == 1 ? var.ssh_key_pair : "${var.generate_ssh_key_pair == "true" ? element(aws_key_pair.ssh.*.key_name, 0) : ""}"}"
+}
+
+output "New ssh keypair generated" {
```

Comment:
use standard `snake_case` style names since outputs can be referenced programmatically by other modules.